### PR TITLE
ExprVectorRandom - Fix Vectors not Unit & Not True Random Generations

### PR DESCRIPTION
**Pull request by DelayedGaming**
### Description
This PR fixes ExprVectorRandom to strictly returning vectors with magnitude of 1.

#### EDIT (24 April)
Permission was granted to change the way vectors are generated within this PR scope.

---
**Target Minecraft Versions:** any
**Requirements:** -
**Related Issues:** -

### CODE REVIEW

File: src/main/java/ch/njol/skript/expressions/ExprVectorRandom.java [hunk 1]
```diff
@@ -36,12 +36,9 @@
 import ch.njol.util.Kleenean;
 import ch.njol.util.coll.CollectionUtils;
 
-/**
- * @author bi0qaw
- */
 @Name("Vectors - Random Vector")
-@Description("Creates a random vector.")
-@Examples({"set {_v} to a random vector"})
+@Description("Creates a random vector with the magnitude of 1.")
```
Comment (sovdeeth) [inline src/main/java/ch/njol/skript/expressions/ExprVectorRandom.java:40]:
```suggestion
@Description("Creates a random unit vector.")
```
Alternatively: `random normalized vector`.

**Review by sovdeeth — APPROVED** [review]

File: src/main/java/ch/njol/skript/expressions/ExprVectorRandom.java (hunk 1 above)
Comment (DelayedGaming) [reply src/main/java/ch/njol/skript/expressions/ExprVectorRandom.java:40]: 'unit vector' seems better. Thanks!

File: src/main/java/ch/njol/skript/expressions/ExprVectorRandom.java [hunk 2]
```diff
@@ -57,8 +54,8 @@ public boolean init(Expression<?>[] exprs, int matchedPattern, Kleenean isDelaye
 	}
 
 	@Override
-	protected Vector[] get(Event e) {
-		return CollectionUtils.array(new Vector(randomSignedDouble(), randomSignedDouble(), randomSignedDouble()));
+	protected Vector[] get(Event event) {
+		return CollectionUtils.array(new Vector(randomSignedDouble(), randomSignedDouble(), randomSignedDouble()).normalize());
```
Comment (sovdeeth) [inline src/main/java/ch/njol/skript/expressions/ExprVectorRandom.java:58]:
```suggestion
		// Generating uniform random numbers leads to bias towards the corners of the cube. Gaussian distribution is radially symmetric, so it's properly uniform.
		return CollectionUtils.array(new Vector(random.nextGaussian(), random.nextGaussian(), random.nextGaussian()).normalize());
```

**Review by sovdeeth — COMMENTED** [review]
This is entirely up to you, but while you're here would you be willing to fix the "not actually random" aspect of this expression? Uniform numbers lead to bias towards the corners of the cube, rather than properly random generation. Using normally distributed components fixes this. 
You'd also have to delete the randomSignedDouble method.

If you don't feel like it, I can always do it in another PR!

**Review by TheLimeGlass — APPROVED** [review]

File: src/main/java/ch/njol/skript/expressions/ExprVectorRandom.java (hunk 2 above)
Comment (DelayedGaming) [reply src/main/java/ch/njol/skript/expressions/ExprVectorRandom.java]: Permission was granted to change the way vectors are generated within this PR scope.

**Review by sovdeeth — APPROVED** [review]

**Review by AyhamAl-Ali — APPROVED** [review]

**Pull request merged**